pkg/cimfs/cim_test.go

```diff
@@ -60,6 +84,99 @@ func createCimFileUtil(c *CimFsWriter, fileTuple tuple) error {
 	return nil
 }
 
+// openNewCIM creates a new CIM and returns a writer to that CIM.  The caller MUST close
+// the writer.
+func openNewCIM(t *testing.T, newCIM testCIM) *CimFsWriter {
+	t.Helper()
+
+	var (
+		writer *CimFsWriter
+		err    error
+	)
+
+	switch val := newCIM.(type) {
+	case *testForkedCIM:
+		writer, err = Create(val.imageDir, val.parentName, val.imageName)
+	case *testBlockCIM:
+		writer, err = CreateBlockCIM(val.BlockPath, val.CimName, val.Type)
+	}
+	if err != nil {
+		t.Fatalf("failed while creating a cim: %s", err)
+	}
+	t.Cleanup(func() {
+		writer.Close()
+		// add 3 second sleep before test cleanup remove the cim directory
+		// otherwise, that removal fails due to some handles still being open
+		time.Sleep(3 * time.Second)
+	})
+	return writer
+}
+
+// compareContent takes in path to a directory (which is usually a volume at which a CIM is
+// mounted) and ensures that every file/directory in the `testContents` shows up exactly
+// as it is under that directory.
+func compareContent(t *testing.T, root string, testContents []tuple) {
+	t.Helper()
+
+	for _, ft := range testContents {
```

Comment:
Should we be checking if there are files/dirs in the cim that are not supposed to be there at all?

Reply:
Yeah, I thought about that. I was looking at some library that would allow me to compare file system trees , but didn't find anything that is directly usable here and adding something like that just for the sake of this would be an overkill, I think. So, for now, just checking for expected files is enough. But I will keep looking for a solution to compare trees. Is that okay?

Reply:
Yes, fine to do this later. My suggestion would be to use `fs.WalkDir` to check the mounted cim and compare to expected content.